compliance/google/long_stopped_instances/google_long_stopped_instances.pt

```diff
@@ -0,0 +1,286 @@
+long_description "Version: 1.0"
+category "Compliance"
+severity "low"
+
+###############################################################################
+# Permissions
+###############################################################################
+
+permission "perm_read_creds" do
+  actions   "rs_cm.show_sensitive","rs_cm.index_sensitive"
+  resources "rs_cm.credentials"
+end
+
+###############################################################################
+# Parameters
+###############################################################################
+
+parameter "param_email" do
+  type "list"
+  label "Email List"
+  description "Email addresses of the recipients you wish to notify"
+end
+parameter "param_exclude_tags" do
+  type "list"
+  label "Tags to ignore"
+  description "List of tags(Key or Key, value mentioned in object metadata.) that will exclude objects from being evaluated by this policy. Multiple tags are evaluated as an 'OR' condition. Tag keys or key/value pairs can be listed. Example: 'test,env=dev'"
+end
+parameter "param_stoped_days" do
+  type "number"
+  label "Stopped days"
+  description "Number of days an instance is stopped before it is added to the report"
+end
+parameter "param_google_project" do
+  type "string"
+  label "Google Cloud Project"
+  description "Google Cloud Project"
+  allowed_pattern /^[0-9a-z:\.-]+$/
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+auth "auth_google", type: "oauth2" do
+  token_url "https://www.googleapis.com/oauth2/v4/token"
+  grant type: "jwt_bearer" do
+    iss cred("GCE_PLUGIN_ACCOUNT")
+    aud "https://www.googleapis.com/oauth2/v4/token"
+    additional_claims do {
+      "scope" => "https://www.googleapis.com/auth/monitoring.write https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/monitoring.read https://www.googleapis.com/auth/monitoring https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/compute.readonly"
+    } end
+    signing_key cred("GCE_PLUGIN_PRIVATE_KEY")
+  end
+end
+
+#https://cloud.google.com/compute/docs/reference/rest/v1/zones/list
+datasource "ds_zone" do
+  request do
+    auth $auth_google
+    host "compute.googleapis.com"
+    path join(["/compute/v1/projects/",$param_google_project,"/zones/"])
+  end
+  result do
+    encoding "json"
+      collect jmes_path(response, "items[*]") do
+      field "zone", jmes_path(col_item, "name")
+    end
+  end
+end
+
+#https://cloud.google.com/compute/docs/reference/rest/v1/instances/list
+datasource "ds_stoped_instances" do
+  iterate $ds_zone
+  request do
+    auth $auth_google
+    host "compute.googleapis.com"
+    path join(["/compute/v1/projects/",$param_google_project,"/zones/",val(iter_item, "zone"),"/instances"])
+    query "filter", 'status="TERMINATED"'
```

Comment:
I don't believe you want terminated, you want a stopped status.    The action won't be able to terminate an already terminated instance.

Reply:
Hi Curt, In case of Google cloud, all the STOPPED instances status is TERMINATED, so I am passing the filter as TERMINATED and able to fetch all the STOPPED instances. And in action I am performing delete operation for the STOPPED/TERMINATED instances.Thanks

Reply:
the terminated status means the instance no longer exists and can not be restarted.  stopped status means the resource is shutdown and can be started again, and stopped again.  you can stop a running instance and it's these instances we need to be checking.

https://cloud.google.com/compute/docs/reference/rest/v1/instances/get 
[Output Only] The status of the instance. One of the following values: PROVISIONING, STAGING, RUNNING, STOPPING, STOPPED, SUSPENDING, SUSPENDED, and TERMINATED.

Reply:
Thank you, Curt, for the update.
I have gone through the API and Google Cloud Instance life cycle. I found the below diagram.
https://cloud.google.com/compute/docs/instances/instance-life-cycle#checking_an_instances_status
![image](https://user-images.githubusercontent.com/55230965/68473384-de60ad00-0248-11ea-8667-26c1ab646bd7.png)

I have also verified the same in google cloud platform and Tested an stopped instance status and got status value TERMINATED. Please check the below screenshots.
 
![image](https://user-images.githubusercontent.com/55230965/68473442-fc2e1200-0248-11ea-9f55-4f46b70578ee.png)

API response.

![image](https://user-images.githubusercontent.com/55230965/68473480-17008680-0249-11ea-8327-450d0ecf841c.png)


Please correct me  if I am wrong.
Thanks,

Reply:
@gopinathmahakud1 thank you for the details.  I also went through the same validation because I was surprised about this result.    I stand corrected, you are correct that the status should be terminated.